client/src/components/Ticketing/ticketingmanager/showings/ShowingUpdated/ShowingUtils.tsx

```diff
@@ -0,0 +1,227 @@
+    })
+    .then((data) => {
+      eventData.eventname = data.data[0].title;
+      eventData.eventdescription = data.data[0].description;
+      eventData.active = data.data[0].active;
+      eventData.imageurl = data.data[0].imageurl;
+      eventData.eventid = data.data[0].eventid;
+      eventData.seasonid_fk = data.data[0].seasonid_fk;
+    });
+  setEventData(eventData);
+};
+
+export const useFetchEventData = (eventID: number) => {
+  const [eventData, setEventData] = useState(undefined);
+  const [loading, setLoading] = useState(true);
+  const [ticketTypes, setTicketTypes] = useState([]);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    void fetchTicketTypes(setTicketTypes);
+    if (eventID) {
+      getEventData(eventID, setEventData)
+        .then(() => console.log('Event load success'))
+        .catch(() => navigate(`/ticketing/showings/v2/${eventID}/notFound`));
+    }
+    setLoading(false);
+  }, [eventID]);
+
+  // eslint-disable-next-line max-len
+  return {eventData, setEventData, loading, ticketTypes};
+};
+
+export const fetchToken = async (getAccessTokenSilently, setToken) => {
+  try {
+    setToken(
+      await getAccessTokenSilently({
+        audience: process.env.REACT_APP_ROOT_URL,
+        scope: 'admin',
+      }),
+    );
+  } catch (error) {
+    console.log(error);
+  }
+};
+export const useFetchToken = () => {
+  const [token, setToken] = useState('');
+  const {getAccessTokenSilently, isAuthenticated} = useAuth0();
+  useEffect(() => {
+    void fetchToken(getAccessTokenSilently, setToken);
+  }, [isAuthenticated]);
+  return {token};
+};
+
+export const useFetchShowingData = (eventID: number) => {
+  const [showingData, setShowingData] = useState([]);
+  const [reload, setReloadShowing] = useState(true);
+
+  useEffect(() => {
+    if (eventID) {
+      void getShowingData(eventID, setShowingData);
+    }
+  }, [eventID, reload]);
+
+  return {showingData, setReloadShowing, reload};
+};
+export const getShowingData = async (eventID, setShowingData) => {
+  try {
+    const showingRes = await fetch(
+      `${process.env.REACT_APP_API_1_URL}/events/instances/${eventID}`,
+    );
+
+    if (!showingRes.ok) {
+      throw new Error('Unable to fetch showings');
+    }
+
+    const data = await showingRes.json();
+    const showingData = data.data;
+
+    for (const showing of showingData) {
+      if (!showing.seatsForType) {
+        showing.seatsForType = [];
+      }
+      if (!showing.ticketTypeId) {
+        showing.ticketTypeId = [];
+      }
+      const ticketRestrictionRes = await fetch(
+        `${process.env.REACT_APP_API_1_URL}/tickets/restrictions/
+          ${showing.eventinstanceid}`,
+      );
+      if (!ticketRestrictionRes.ok) {
+        throw new Error(`Unable to fetch ticket restrictions for ${showing.eventinstanceid}`);
+      }
+      const ticketRestrictionData = await ticketRestrictionRes.json();
+      for (const item of ticketRestrictionData.data) {
```

Comment:
For readability in my opinion, using JS's higher order `forEach` can benefit here since its a bit more self-documenting. Something like 

```
  ticketRestrictionData.data.forEach((item) => {
        showing.seatsForType.push(item.ticketlimit);
        showing.ticketTypeId.push(item.tickettypeid_fk);
  });
```

Reply:
Updated